src/Stryker.Core/Stryker.Core/Mutators/PatternMutatorBase.cs

```diff
@@ -5,73 +5,72 @@
+    {
+        [SyntaxKind.OrPattern] = new[] { SyntaxKind.AndPattern },
+        [SyntaxKind.AndPattern] = new[] { SyntaxKind.OrPattern },
+        [SyntaxKind.LessThanEqualsToken] = new[] { SyntaxKind.GreaterThanToken, SyntaxKind.LessThanToken },
+        [SyntaxKind.GreaterThanEqualsToken] = new[] { SyntaxKind.GreaterThanToken, SyntaxKind.LessThanToken },
+        [SyntaxKind.LessThanToken] = new[] { SyntaxKind.GreaterThanToken, SyntaxKind.LessThanEqualsToken },
+        [SyntaxKind.GreaterThanToken] = new[] { SyntaxKind.LessThanToken, SyntaxKind.GreaterThanEqualsToken }
+    };
 
-        protected PatternMutatorBase() => KindsToMutate = new()
-        {
-            [SyntaxKind.AndPattern] = new[] { SyntaxKind.OrPattern },
-            [SyntaxKind.OrPattern] = new[] { SyntaxKind.AndPattern },
-            [SyntaxKind.LessThanEqualsToken] = new[] { SyntaxKind.GreaterThanToken, SyntaxKind.LessThanToken },
-            [SyntaxKind.GreaterThanEqualsToken] = new[] { SyntaxKind.GreaterThanToken, SyntaxKind.LessThanToken },
-            [SyntaxKind.LessThanToken] = new[] { SyntaxKind.GreaterThanToken, SyntaxKind.LessThanEqualsToken },
-            [SyntaxKind.GreaterThanToken] = new[] { SyntaxKind.LessThanToken, SyntaxKind.GreaterThanEqualsToken }
-        };
+    /// <summary> Apply mutations to a <see cref="PatternSyntax"/></summary>
+    protected IEnumerable<Mutation> ApplyMutations(PatternSyntax node, SemanticModel semanticModel) => node switch
+    {
+        BinaryPatternSyntax binaryPattern => ApplyMutations(binaryPattern),
+        RelationalPatternSyntax relationalPattern => ApplyMutations(relationalPattern),
+        _ => Enumerable.Empty<Mutation>()
+    };
 
-        /// <summary> Apply mutations to a <see cref="PatternSyntax"/></summary>
-        protected IEnumerable<Mutation> ApplyMutations(PatternSyntax node, SemanticModel semanticModel) => node switch
+    private IEnumerable<Mutation> ApplyMutations(BinaryPatternSyntax node)
+    {
+        if (!KindsToMutate.TryGetValue(node.Kind(), out var mutations))
         {
-            BinaryPatternSyntax binaryPattern => ApplyMutations(binaryPattern),
-            RelationalPatternSyntax relationalPattern => ApplyMutations(relationalPattern),
-            _ => Enumerable.Empty<Mutation>()
-        };
+            yield break;
+        }
 
-        private IEnumerable<Mutation> ApplyMutations(BinaryPatternSyntax node)
+        foreach (var mutation in mutations)
         {
-            if (KindsToMutate.TryGetValue(node.Kind(), out var mutations))
+            // can't usage the update method here, because roslyn implementation is broken
+            var replacementNode = SyntaxFactory.BinaryPattern(mutation, node.Left, node.Right);
+            replacementNode = replacementNode.WithOperatorToken(replacementNode.OperatorToken.WithTriviaFrom(node.OperatorToken));
+            replacementNode.Update(node.Left, node.OperatorToken, node.Right);
```

Comment:
You commented that you couldn't use the `update` method here. Does this still work then?